### PR TITLE
[COM-26][COM-113][COM-114][COM-115][COM-166][COM-117] Camera Component, Camera Entity, Entity Manager Statically Available, Scene Definition, Scenes have root notes, EntityManager can Instantiate to Scene, Scene Description, Entity Components and Transforms symbiosis

### PR DESCRIPTION
**Pull request by le-rich**

### CODE REVIEW

File: Core/include/Utils/IDUtils.h [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #pragma once
 
-#include "../../External/uuid/uuid_v4.h"
 #include "../../External/uuid/uuid_v4.h"
 
-UUIDv4::UUIDGenerator<std::mt19937_64> uuidGen;
\ No newline at end of file
+extern UUIDv4::UUIDGenerator<std::mt19937_64> uuidGen;
```
Comment (Tissure) [inline Core/include/Utils/IDUtils.h:5]: Add comment explaining what `extern` is and why we should be conscious of adding it when needed.

File: Core/include/Managers/EntityManager.h [hunk 2]
```diff
@@ -81,15 +98,54 @@ class EntityManager{
...
+        UUIDv4::UUID parentUUID = parent->GetEntityID();
+
+        getEntity(parentUUID)->addChild(entity);
+        entity->setParent(parent);
+
+        entity->transform = transform;
 
+        addEntityToMap(*entity);
     }
 };
```
Comment (Tissure) [inline Core/include/Managers/EntityManager.h:151]: Add ticket/TODO for removing entities

**Review by Tissure — CHANGES_REQUESTED** [review]
pwease add the wequested cowomments and owopshiuwunally wite up owon the fwixes
~uwu~ 😘😘😘

File: Core/include/Components/Transform.h [hunk 3]
```diff
@@ -65,6 +65,6 @@ class Transform {
 
 	glm::vec3 getLocalScale();
 	void setLocalScale(glm::vec3 scale);
-};
 
-#endif 
\ No newline at end of file
+	Transform& operator=(const Transform& other);
```
Comment (Tissure) [inline Core/include/Components/Transform.h:69]: Oops guess we forgot we needed these. Part of optional write up.

File: Core/include/Managers/EntityManager.h [hunk 4]
```diff
@@ -10,18 +13,29 @@ class EntityManager{
 protected:
     static const int MAX_ENTITIES = 3000;
     // TODO: map of entities
-    std::map<UUIDv4::UUID, Entity> entityMap;
+    std::map<UUIDv4::UUID, Entity*> entityMap;
 
-public:
-    // default constructor
+    // Private constructor as this should be a singleton.
```
[the remark this answers is not in the document]
Comment (le-rich) [reply Core/include/Managers/EntityManager.h:18]: I don't see a benefit to that. If you have to flush and delete a previous hierarchy only to spin up a new one wouldnt that be the same as keeping the same object and flushing its entity map.

What if we need to keep entities between scenes such as game manager script components.

Scene itself doest have any tree other than storing a referece to the root node, if annything entitymanager is the defacto hierarchy, with entities that cann exist between scenes.

File: Core/include/Managers/EntityManager.h (hunk 2 above)
Comment (le-rich) [reply Core/include/Managers/EntityManager.h:151]: Ticket created

Comment (le-rich) [reply Core/include/Managers/EntityManager.h:151]: COM-123

File: Core/include/Utils/IDUtils.h (hunk 1 above)
Comment (le-rich) [reply Core/include/Utils/IDUtils.h:5]: I dont like adding verbose comments to code, people should be googling extern and its use.

Comment (Tissure) [reply Core/include/Utils/IDUtils.h:5]: Understandable

**Review by Tissure — APPROVED** [review]
uwu

**Pull request merged**